iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/digitaltwin/DigitalTwinClientTests.java

```diff
@@ -108,12 +128,51 @@ public static void cleanUpAfterClass()
         registryManager.close();
     }
 
+    @BeforeClass
+    public static void startProxy()
+    {
+        proxyServer = DefaultHttpProxyServer.bootstrap()
+            .withPort(testProxyPort)
+            .start();
+    }
+
+    @AfterClass
+    public static void stopProxy()
+    {
+        proxyServer.stop();
+    }
+
     @Test
     @StandardTierHubOnlyTest
     public void getDigitalTwin() {
         // act
         BasicDigitalTwin response = digitalTwinClient.getDigitalTwin(deviceId, BasicDigitalTwin.class);
-        ServiceResponseWithHeaders<BasicDigitalTwin, DigitalTwinGetHeaders> responseWithHeaders = digitalTwinClient.getDigitalTwinWithResponse(deviceId, BasicDigitalTwin.class);
+        ServiceResponseWithHeaders<BasicDigitalTwin, DigitalTwinGetHeaders> responseWithHeaders =
+            digitalTwinClient.getDigitalTwinWithResponse(deviceId, BasicDigitalTwin.class);
+
+        // assert
+        assertEquals(response.getMetadata().getModelId(), E2ETestConstants.THERMOSTAT_MODEL_ID);
+        assertEquals(responseWithHeaders.body().getMetadata().getModelId(), E2ETestConstants.THERMOSTAT_MODEL_ID);
+    }
+
+    @Test
+    @StandardTierHubOnlyTest
+    public void getDigitalTwinWithProxy() {
+        // arrange
+        Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(testProxyHostname, testProxyPort));
+        ProxyOptions proxyOptions = new ProxyOptions(proxy);
+        DigitalTwinClientOptions clientOptions =
+            DigitalTwinClientOptions.builder()
+                .proxyOptions(proxyOptions)
+                .httpReadTimeout(0)
```

Comment:
any test to verify that these timeout values are honored?

Reply:
It's tricky to test this since we can't just make an HTTP call and hope that it will time out in even 1 millisecond. I can add some tests though that ensure that a negative timeout value throws, since that should show that the rest client is getting these values

Reply:
That's fair, we don't need to test their http client, just that they acknowledge our custom values :)